File: frontend/src/lib/estoque.ts
// Mesmo shape de backend/src/estoque/dto/estoque-response.dto.ts
// (EstoqueItemDto/EstoqueConsultaDto) - duplicado aqui por não haver
// pacote compartilhado entre front e back. Estoque.svc (fonte atual, ver
// comentário no DTO do backend) devolve saldo CONSOLIDADO por produto, sem
// quebra por local de estocagem - localCodigo/localNome/lote/fabricadoEm
// ficam sempre null por enquanto, `itens` nunca passa de 1 elemento (o
// shape de lista foi mantido só pra não quebrar front/mobile).
export interface EstoqueItemDto {
  localCodigo: string | null;
  localNome: string | null;
  lote: string | null;
  fabricadoEm: string | null;
  quantidade: string;
}

export interface EstoqueConsultaDto {
  produtoId: string;
  codigo: string;
  itens: EstoqueItemDto[];
  // Momento da última sincronização bem-sucedida do saldo (não da consulta
  // em si) - null quando o produto existe mas nunca teve saldo sincronizado.
  atualizadoEm: string | null;
}

// Mesmo shape de backend/src/estoque/dto/estoque-mais-pedidos.dto.ts
// (ProdutoMaisPedidoDto, GET /estoque/mais-pedidos) - ranking por
// quantidade total pedida (não valor), pra priorizar reposição de
// estoque.
export interface ProdutoMaisPedidoDto {
  produtoId: string;
  nome: string | null;
  codigo: string;
  quantidadeTotalPedida: number;
  quantidadeDisponivel: string | null;
}

File: mobile/lib/core/models/estoque.dart
/// Mesmo shape de `backend/src/estoque/dto/estoque-response.dto.ts` -
/// duplicado aqui por não haver pacote compartilhado entre mobile e back.
class EstoqueItem {
  const EstoqueItem({
    required this.localCodigo,
    required this.localNome,
    required this.lote,
    required this.fabricadoEm,
    required this.quantidade,
  });

  factory EstoqueItem.fromJson(Map<String, dynamic> json) {
    return EstoqueItem(
      localCodigo: json['localCodigo'] as String?,
      localNome: json['localNome'] as String?,
      lote: json['lote'] as String?,
      fabricadoEm: json['fabricadoEm'] as String?,
      quantidade: json['quantidade'] as String,
    );
  }

  final String? localCodigo;
  final String? localNome;
  final String? lote;
  final String? fabricadoEm;
  final String quantidade;

  String get tituloLocal => localNome ?? localCodigo ?? 'Local não identificado';
}

class ResultadoEstoque {
  const ResultadoEstoque({required this.produtoId, required this.codigo, required this.itens});

  factory ResultadoEstoque.fromJson(Map<String, dynamic> json) {
    return ResultadoEstoque(
      produtoId: json['produtoId'] as String,
      codigo: json['codigo'] as String,
      itens: (json['itens'] as List)
          .cast<Map<String, dynamic>>()
          .map(EstoqueItem.fromJson)
          .toList(),
    );
  }

  final String produtoId;
  final String codigo;
  final List<EstoqueItem> itens;
}
